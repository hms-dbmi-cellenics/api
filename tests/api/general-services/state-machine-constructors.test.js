const constants = require('../../../src/api/general-services/pipeline-manage/constants');
const fake = require('../../test-utils/constants');
const { buildPodRequest } = require('../../../src/api/general-services/pipeline-manage/constructors/assign-pod-to-pipeline');
const validateRequest = require('../../../src/utils/schema-validator');

describe('Test for pipeline constructor services', () => {
  it('builds valid pipeline pod assignation request', async () => {
    const message = buildPodRequest(fake.SANDBOX_ID,
      fake.EXPERIMENT_ID,
      constants.ASSIGN_POD_TO_PIPELINE,
      constants.GEM2S_PROCESS_NAME,
      fake.ACTIVITY_ID);

    await validateRequest(message, 'PipelinePodRequest.v1.yaml');
    expect(message).toMatchSnapshot();
  });
});
