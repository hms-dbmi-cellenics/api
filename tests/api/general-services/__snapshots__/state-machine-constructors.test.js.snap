// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for pipeline constructor services builds valid pipeline pod assignation request 1`] = `
Object {
  "experimentId": "experimentid11111111111111111111",
  "input": Object {
    "activityId": "arn:aws:states:eu-east-9:111111111111:activity:biomage-gem2s-production-39249897-cfce-632b-a617-e58fbf251733",
    "experimentId": "experimentid11111111111111111111",
    "processName": "gem2s",
    "sandboxId": "default",
  },
  "taskName": "assignPodToPipeline",
}
`;
